Core/Services/StitchCoreRemoteMongoDBService/Tests/StitchCoreRemoteMongoDBServiceTests/Sync/ChangeEventUnitTests.swift

```diff
@@ -0,0 +1,112 @@
+            ns: namespace,
+            documentKey: ["_id": docId],
+            updateDescription: nil,
+            hasUncommittedWrites: writePending)
+
+        let actualLocalInsertEvent = ChangeEvent<Document>.changeEventForLocalInsert(namespace: namespace,
+                                                                                     document: document,
+                                                                                     writePending: writePending)
+        XCTAssertEqual(expectedLocalInsertEvent, actualLocalInsertEvent)
+    }
+
+    func testChangeEventForLocalReplace() {
+        let expectedLocalReplaceEvent = ChangeEvent<Document>(
+            id: Document(),
+            operationType: .replace,
+            fullDocument: document,
+            ns: namespace,
+            documentKey: ["_id": docId],
+            updateDescription: nil,
+            hasUncommittedWrites: writePending)
+
+        let actualLocalReplaceEvent = ChangeEvent<Document>.changeEventForLocalReplace(namespace: namespace, documentId: docId, document: document, writePending: writePending)
+
+        XCTAssertEqual(expectedLocalReplaceEvent, actualLocalReplaceEvent)
+    }
+
+    func testChangeEventForLocalUpdate() {
+        let expectedLocalUpdateEvent = ChangeEvent<Document>(
+            id: Document(),
+            operationType: .update,
+            fullDocument: document,
+            ns: namespace,
+            documentKey: ["_id": docId],
+            updateDescription: UpdateDescription.init(updatedFields: ["foo": 42,
+                                                                      "bar": "baz"],
+                                                      removedFields: []),
+            hasUncommittedWrites: writePending)
+
+        let actualLocalUpdateEvent = ChangeEvent<Document>.changeEventForLocalUpdate(
+            namespace: namespace,
+            documentId: docId,
+            update: UpdateDescription.init(updatedFields: ["foo": 42, "bar": "baz"], removedFields: []),
+            fullDocumentAfterUpdate: document,
+            writePending: writePending)
+
+        XCTAssertEqual(expectedLocalUpdateEvent, actualLocalUpdateEvent)
+    }
+
+    func testChangeEventForLocalDelete() {
+        let expectedLocalDeleteEvent = ChangeEvent<Document>(
+            id: Document(),
+            operationType: .delete,
+            fullDocument: nil,
+            ns: namespace,
+            documentKey: ["_id": docId],
+            updateDescription: nil,
+            hasUncommittedWrites: writePending)
+
+        let actualLocalDeleteEvent = ChangeEvent<Document>.changeEventForLocalDelete(namespace: namespace,
+                                                                                     documentId: docId,
+                                                                                     writePending: writePending)
+
+        XCTAssertEqual(expectedLocalDeleteEvent, actualLocalDeleteEvent)
+    }
+}
```

Comment:
[major] We should pin the error behavior for unknown with a test to make sure that all exception states are deliberate rather than implicit